test/test.tools.js

```diff
@@ -51,36 +61,53 @@ describe("Monkeypatching", () => {
 describe("Test tools for parsing new gmail body_params", () => {
     const gmail = new Gmail();
     const data =  JSON.parse(testData.new_gmail_archive_action_body_params);
+    const threads = gmail.tools.extract_from_graph(data, gmail.check.data.is_thread);
+    const threadData = threads.map(thread => gmail.tools.get_thread_data(thread))[0];
 
     it("get thread id", () => {
-        const thread = gmail.tools.get_thread_id(data);
+        const thread = gmail.tools.get_thread_id(threads[0]);
 
```

Comment:
You've changed to parameter from `data` to `threads[0]`... Any particular reason for that?

Reply:
I changed data in `testdata-parser.js` and now we have clear data, the same as sending us gmail )

Reply:
Right. In which case that's technically an improvement. Good good 😃 

That said, we're changing things in several places here, and as a result I'm struggling to keep full context of the code.

Wouldn't it make more sense to just modify the base test-data setup instead of every where it's used then? Like this?

````diff
-const data =  JSON.parse(testData.new_gmail_archive_action_body_params);
+const xhrData =  JSON.parse(testData.new_gmail_archive_action_body_params);
+const threads = gmail.tools.extract_from_graph(data, gmail.check.data.is_thread);
+const data = threads[0];
````

And then we can leave all the existing tests as is to minimize the diff and get focus on the "important" stuff?

Reply:
I think prev test data did not meet the requirements we have good test data for other events now. 
In this line we prepare data
`const data =  JSON.parse(testData.new_gmail_archive_action_body_params);`
in this we prepare threads from clear gmail response
`const threads = gmail.tools.extract_from_graph(data, gmail.check.data.is_thread);`

Maybe in future, we need to refactor this code for another tool but now it looks clearly.

What do you think?)